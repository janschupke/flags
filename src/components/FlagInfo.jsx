import React from 'react';
import {
  InfoSection,
  InfoCaption,
  InfoGrid,
  InfoItem,
  InfoLabel,
  InfoValue,
  InfoPlaceholder,
  InfoWrapper
} from './styled/FlagQuiz.styles';

const FlagInfo = ({ prev }) => {
  return (
    <InfoSection>
      <InfoWrapper>
        <InfoCaption>Previous Flag Information</InfoCaption>
        {prev ? (
          <InfoGrid>
            <InfoItem>
              <InfoLabel>Country</InfoLabel>
              <InfoValue>{prev.name || 'N/A'}</InfoValue>
            </InfoItem>
            <InfoItem>
              <InfoLabel>Capital</InfoLabel>
              <InfoValue>{prev.capital || 'N/A'}</InfoValue>
            </InfoItem>
            <InfoItem>
              <InfoLabel>Continent</InfoLabel>
              <InfoValue>{prev.continent || 'N/A'}</InfoValue>
            </InfoItem>
            <InfoItem>
              <InfoLabel>Government</InfoLabel>
              <InfoValue>{prev.government || 'N/A'}</InfoValue>
            </InfoItem>
            <InfoItem>
              <InfoLabel>Languages</InfoLabel>
              <InfoValue>{prev.language || 'N/A'}</InfoValue>
            </InfoItem>
            <InfoItem>
              <InfoLabel>Your Answer</InfoLabel>
              <InfoValue>{prev.user || 'N/A'}</InfoValue>
            </InfoItem>
          </InfoGrid>
        ) : (
          <InfoPlaceholder>
            No previous flag information yet.<br />Submit your first answer to see details here.
          </InfoPlaceholder>
        )}
      </InfoWrapper>
    </InfoSection>
  );
};

export default FlagInfo; 
