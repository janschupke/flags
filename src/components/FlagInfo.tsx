import React from 'react';
import {
  InfoSection,
  InfoCaption,
  InfoGrid,
  InfoItem,
  InfoLabel,
  InfoValue,
  InfoPlaceholder,
  InfoWrapper
} from './styled/FlagQuiz.styles';

interface FlagInfoProps {
  prev: {
    name?: string;
    capital?: string;
    continent?: string;
    government?: string;
    language?: string;
    user?: string;
    isCorrect?: boolean;
  } | null;
}

const FlagInfo: React.FC<FlagInfoProps> = ({ prev }) => {
  return (
    <InfoSection>
      <InfoWrapper>
      <InfoCaption>Previous Country Information</InfoCaption>
      {prev ? (
        <InfoGrid>
          <InfoItem>
            <InfoLabel>Name:</InfoLabel>
            <InfoValue>{prev.name || 'Not available'}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>Capital:</InfoLabel>
            <InfoValue>{prev.capital || 'Not available'}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>Continent:</InfoLabel>
            <InfoValue>{prev.continent || 'Not available'}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>Government:</InfoLabel>
            <InfoValue>{prev.government || 'Not available'}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>Language:</InfoLabel>
            <InfoValue>{prev.language || 'Not available'}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>Your Answer:</InfoLabel>
            <InfoValue>{prev.user || 'Not available'}</InfoValue>
          </InfoItem>
        </InfoGrid>
      ) : (
        <InfoPlaceholder>
          No previous flag information yet.<br />Submit your first answer to see details here.
        </InfoPlaceholder>
      )}
      </InfoWrapper>
    </InfoSection>
  );
};

export default FlagInfo; 
