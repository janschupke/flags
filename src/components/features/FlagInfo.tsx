import React from 'react';
import {
  InfoSection,
  InfoCaption,
  InfoGrid,
  InfoItem,
  InfoLabel,
  InfoValue,
  InfoPlaceholder,
  InfoWrapper
} from '../styled/FlagQuiz.styles';
import { FlagInfoProps } from '../../types';
import { UI_TEXT } from '../../constants';

const FlagInfo: React.FC<FlagInfoProps> = ({ prev }) => {
  return (
    <InfoSection>
      <InfoWrapper>
      <InfoCaption>{UI_TEXT.previousCountryInfo}</InfoCaption>
      {prev ? (
        <InfoGrid>
          <InfoItem>
            <InfoLabel>{UI_TEXT.labels.name}</InfoLabel>
            <InfoValue>{prev.name || UI_TEXT.fallback}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>{UI_TEXT.labels.capital}</InfoLabel>
            <InfoValue>{prev.capital || UI_TEXT.fallback}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>{UI_TEXT.labels.continent}</InfoLabel>
            <InfoValue>{prev.continent || UI_TEXT.fallback}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>{UI_TEXT.labels.government}</InfoLabel>
            <InfoValue>{prev.government || UI_TEXT.fallback}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>{UI_TEXT.labels.language}</InfoLabel>
            <InfoValue>{prev.language || UI_TEXT.fallback}</InfoValue>
          </InfoItem>
          <InfoItem>
            <InfoLabel>{UI_TEXT.labels.yourAnswer}</InfoLabel>
            <InfoValue>{prev.user || UI_TEXT.fallback}</InfoValue>
          </InfoItem>
        </InfoGrid>
      ) : (
        <InfoPlaceholder>
          {UI_TEXT.placeholder.previousInfo}
        </InfoPlaceholder>
      )}
      </InfoWrapper>
    </InfoSection>
  );
};

export default FlagInfo; 
