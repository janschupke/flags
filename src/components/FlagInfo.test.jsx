import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import FlagInfo from './FlagInfo';

describe('FlagInfo', () => {
  it('shows placeholders for previous info on first render', () => {
    render(<FlagInfo prev={null} />);
    expect(screen.getByText(/no previous flag information yet/i)).toBeInTheDocument();
  });

  it('populates info after answer', () => {
    render(<FlagInfo prev={{
      name: 'Finland',
      capital: 'Helsinki',
      continent: 'Europe',
      government: 'Parliamentary republic',
      language: 'Finnish, Swedish',
      user: 'fin',
      isCorrect: true
    }} />);
    expect(screen.getByText(/finland/i)).toBeInTheDocument();
    expect(screen.getByText(/helsinki/i)).toBeInTheDocument();
    expect(screen.getByText(/europe/i)).toBeInTheDocument();
    expect(screen.getByText(/parliamentary republic/i)).toBeInTheDocument();
    expect(screen.getByText(/finnish, swedish/i)).toBeInTheDocument();
    // Use a more specific query for the user answer
    const userAnswerElements = screen.getAllByText(/fin/i);
    expect(userAnswerElements.length).toBeGreaterThan(0);
    // Check that one of them is the user answer
    const userAnswerElement = userAnswerElements.find(el => 
      el.textContent === 'fin' || el.textContent.includes('fin')
    );
    expect(userAnswerElement).toBeTruthy();
  });

  it('shows N/A for missing fields', () => {
    render(<FlagInfo prev={{ name: '', capital: '', continent: '', government: '', language: '', user: '', isCorrect: undefined }} />);
    expect(screen.getAllByText('N/A').length).toBeGreaterThan(2);
  });

  it('handles special characters', () => {
    render(<FlagInfo prev={{ name: "Côte d'Ivoire", capital: "Yamoussoukro", continent: "Africa", government: "Presidential republic", language: "French", user: "côte", isCorrect: false }} />);
    // There are two elements with 'côte', so use getAllByText and check both
    const matches = screen.getAllByText(/côte/i);
    expect(matches.length).toBeGreaterThan(1);
    expect(matches[0].textContent.toLowerCase()).toContain("côte");
    expect(matches[1].textContent.toLowerCase()).toContain("côte");
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<FlagInfo prev={{
      name: 'Finland',
      capital: 'Helsinki',
      continent: 'Europe',
      government: 'Parliamentary republic',
      language: 'Finnish, Swedish',
      user: 'fin',
      isCorrect: true
    }} />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 
