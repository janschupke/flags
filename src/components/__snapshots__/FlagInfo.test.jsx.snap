// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FlagInfo > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-dntSTA jYaWkY"
  >
    <div
      class="sc-gLaqbQ bFmYnO"
    >
      <div
        class="sc-hwddKA iqtixn"
      >
        Previous Flag Information
      </div>
      <div
        class="sc-kvnevz fbqFOh"
      >
        <div
          class="sc-dFVmKS ggAVYB"
        >
          <span
            class="sc-kCuUfV gKEvLf"
          >
            Country
          </span>
          <span
            class="sc-dNdcvo hMNEOk"
          >
            Finland
          </span>
        </div>
        <div
          class="sc-dFVmKS ggAVYB"
        >
          <span
            class="sc-kCuUfV gKEvLf"
          >
            Capital
          </span>
          <span
            class="sc-dNdcvo hMNEOk"
          >
            Helsinki
          </span>
        </div>
        <div
          class="sc-dFVmKS ggAVYB"
        >
          <span
            class="sc-kCuUfV gKEvLf"
          >
            Continent
          </span>
          <span
            class="sc-dNdcvo hMNEOk"
          >
            Europe
          </span>
        </div>
        <div
          class="sc-dFVmKS ggAVYB"
        >
          <span
            class="sc-kCuUfV gKEvLf"
          >
            Government
          </span>
          <span
            class="sc-dNdcvo hMNEOk"
          >
            Parliamentary republic
          </span>
        </div>
        <div
          class="sc-dFVmKS ggAVYB"
        >
          <span
            class="sc-kCuUfV gKEvLf"
          >
            Languages
          </span>
          <span
            class="sc-dNdcvo hMNEOk"
          >
            Finnish, Swedish
          </span>
        </div>
        <div
          class="sc-dFVmKS ggAVYB"
        >
          <span
            class="sc-kCuUfV gKEvLf"
          >
            Your Answer
          </span>
          <span
            class="sc-dNdcvo hMNEOk"
          >
            fin
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
